pkg/sql/plan2/function/operator/cast.go

```diff
@@ -0,0 +1,930 @@
+	resultTyp := lv.Typ
+	lvs := lv.Col.([]types.Decimal128)
+	if lv.IsScalar() {
+		vec := proc.AllocScalarVector(resultTyp)
+		rs := make([]types.Decimal128, 1)
+		copy(rs, lvs)
+		nulls.Set(vec.Nsp, lv.Nsp)
+		vector.SetCol(vec, rs)
+		return vec, nil
+	}
+	vec, err := proc.AllocVector(resultTyp, int64(resultTyp.Size)*int64(len(lvs)))
+	if err != nil {
+		return nil, err
+	}
+	rs := encoding.DecodeDecimal128Slice(vec.Data)
+	rs = rs[:len(lvs)]
+	copy(rs, lvs)
+	nulls.Set(vec.Nsp, lv.Nsp)
+	vector.SetCol(vec, rs)
+	return vec, nil
+}
+
+//  castTimeStampAsDatetime : Cast converts timestamp to datetime decimal128
+func castTimeStampAsDatetime(lv, rv *vector.Vector, proc *process.Process) (*vector.Vector, error) {
+	rtl := 8
+	lvs := lv.Col.([]types.Timestamp)
+	if lv.IsScalar() {
+		vec := proc.AllocScalarVector(rv.Typ)
+		rs := make([]types.Datetime, 1)
+		if _, err := typecast.TimestampToDatetime(lvs, rs); err != nil {
+			return nil, err
+		}
+		nulls.Set(vec.Nsp, lv.Nsp)
+		vector.SetCol(vec, rs)
+		return vec, nil
+	}
+
+	vec, err := proc.AllocVector(rv.Typ, int64(rtl)*int64(len(lvs)))
+	if err != nil {
+		return nil, err
+	}
+	rs := encoding.DecodeDatetimeSlice(vec.Data)
+	rs = rs[:len(lvs)]
+	if _, err := typecast.TimestampToDatetime(lvs, rs); err != nil {
+		return nil, err
+	}
+	nulls.Set(vec.Nsp, lv.Nsp)
+	vector.SetCol(vec, rs)
+	return vec, nil
+}
+
+//  isInteger return true if the types.T is integer type
+func isInteger(t types.T) bool {
+	if t == types.T_int8 || t == types.T_int16 || t == types.T_int32 || t == types.T_int64 ||
+		t == types.T_uint8 || t == types.T_uint16 || t == types.T_uint32 || t == types.T_uint64 {
+		return true
+	}
+	return false
+}
+
+//  isSignedInteger: return true if the types.T is Signed integer type
+func isSignedInteger(t types.T) bool {
+	if t == types.T_int8 || t == types.T_int16 || t == types.T_int32 || t == types.T_int64 {
+		return true
+	}
+	return false
+}
+
+//  isUnsignedInteger: return true if the types.T is UnSigned integer type
+func isUnsignedInteger(t types.T) bool {
+	if t == types.T_uint8 || t == types.T_uint16 || t == types.T_uint32 || t == types.T_uint64 {
+		return true
+	}
+	return false
+}
+
+//  isFloat: return true if the types.T is floating Point Types
+func isFloat(t types.T) bool {
+	if t == types.T_int32 || t == types.T_int64 {
+		return true
+	}
+	return false
+}
+
+//  isNumeric: return true if the types.T is numbric type
+func isNumeric(t types.T) bool {
+	if isInteger(t) || isFloat(t) {
+		return true
+	}
+	return false
+}
+
+//  isString: return true if the types.T is string type
+func isString(t types.T) bool {
+	if t == types.T_char || t == types.T_varchar {
+		return true
+	}
+	return false
+}
+
+//  isDateSeries: return true if the types.T is date related type
+func isDateSeries(t types.T) bool {
+	if t == types.T_date || t == types.T_datetime || t == types.T_datetime {
```

Comment:
static check error
`redundant or: t == types.T_datetime || t == types.T_datetime`

Reply:
OK, think you,I see. I have modified and submitted.